code/modules/mob/living/simple_animal/hostile/thing.dm

```diff
@@ -0,0 +1,2046 @@
+			upgrading = 0
+		else if((upgrade == THING_WEAPONIZE) && (points >= 5))
+			can_weaponize = 1
+			points -= 5
+			to_chat(src, "<span class='warning'>Bought the weaponize ability. You can now extend a dangerous arm blade, allowing you to deal more damage and force open doors.</span>")
+			upgrading = 0
+			var/datum/action/innate/thing/weaponize/weaponize_action = new()
+			weaponize_action.Grant(src)
+		else if((upgrade == THING_CONVERT) && (points >= 10))
+			can_convert = 1
+			points -= 10
+			to_chat(src, "<span class='warning'>Bought the convert ability. Every time you absorb someone, they will be converted into a terrible monster that will follow your orders.</span>")
+			upgrading = 0
+			var/datum/action/innate/thing/communicate/communicate_action = new()
+			communicate_action.Grant(src)
+		else if((upgrade == THING_OMNIVISION) && (points >= 5))
+			can_omnivision = 1
+			update_sight()
+			points -= 5
+			to_chat(src, "<span class='warning'>Bought the omnivision ability. You can now see everything within your view.</span>")
+			upgrading = 0
+		else if((upgrade == THING_SCREECH) && (points >= 5))
+			can_screech = 1
+			points -= 5
+			to_chat(src, "<span class='warning'>Bought the screech ability. You can now let out a screech that will freeze living beings in place and make them release anything they are holding.</span>")
+			upgrading = 0
+			var/datum/action/innate/thing/screech/screech_action = new()
+			screech_action.Grant(src)
+		else if((upgrade == THING_SCREECH_UPGRADE) && (points >= 10))
+			screech_level = 2
+			points -= 10
+			to_chat(src, "<span class='warning'>Upgraded the screech ability. It is now powerful enough to knock living beings down and deafen them.</span>")
+			upgrading = 0
+		else if((upgrade == THING_EMP) && (points >= 5))
+			can_emp = 1
+			points -= 5
+			to_chat(src, "<span class='warning'>Bought the electromagnetic disturbance ability. You can now release a powerful electric blast which will deactivate and break any light source.</span>")
+			upgrading = 0
+			var/datum/action/innate/thing/empscreech/empscreech_action = new()
+			empscreech_action.Grant(src)
+		else if((upgrade == THING_EMP_UPGRADE) && (points >= 10))
+			emp_level = 2
+			points -= 10
+			to_chat(src, "<span class='warning'>Upgraded the electromagnetic disturbance ability. It now also acts as an EMP.</span>")
+			upgrading = 0
+		else if((upgrade == THING_BUILD) && (points >= 10))
+			can_build = 1
+			points -= 10
+			to_chat(src, "<span class='warning'>Bought the build ability. You can now lay creep and construct useful buildings.</span>")
+			upgrading = 0
+			var/datum/action/innate/thing/plantcreep/plantcreep_action = new()
+			plantcreep_action.Grant(src)
+			var/datum/action/innate/thing/build/build_action = new()
+			build_action.Grant(src)
+			var/datum/action/innate/thing/crawlerrally/rally_action = new()
+			rally_action.Grant(src)
```

Comment:
You are checking so many conditions and changing so many variables here that I doubt anyone can really understand what is going on.

This is way too overcomplicated.

Why are such unwieldy strings used as a bitflag macro. Why. Juse use a variable. 

While Crazy's suggestions make it better than comparing plain string the fact such string were used in the first place is problematic.